tfjs-backend-webgpu/src/buffer_manager.ts

```diff
@@ -26,69 +26,63 @@ export class BufferManager {
 
   constructor(private device: GPUDevice) {}
 
-  // The upload buffer is not reusable.
-  acquireUploadBuffer(size: number, usage: GPUBufferUsageFlags) {
-    this.numBytesAllocated += size;
-    return this.device.createBuffer({size, usage, mappedAtCreation: true});
-  }
-
   acquireBuffer(
-      size: number, usage: GPUBufferUsageFlags, mappedAtCreation = false) {
+      size: number, usage: GPUBufferUsageFlags, mappedAtCreation = false,
+      reuse = true) {
+    let buffer;
     const key = getBufferKey(size, usage);
-    if (!this.freeBuffers.has(key)) {
-      this.freeBuffers.set(key, []);
+
+    if (reuse) {
+      if (!this.freeBuffers.has(key)) {
+        this.freeBuffers.set(key, []);
+      }
+
+      if (this.freeBuffers.get(key).length > 0) {
+        buffer = this.freeBuffers.get(key).shift();
+        this.numFreeBuffers--;
+      } else {
+        buffer = this.device.createBuffer({size, usage, mappedAtCreation});
+        this.numBytesAllocated += size;
+      }
+    } else {
+      buffer = this.device.createBuffer({size, usage, mappedAtCreation});
+      this.numBytesAllocated += size;
     }
 
     if (!this.usedBuffers.has(key)) {
       this.usedBuffers.set(key, []);
     }
-
-    this.numBytesUsed += size;
+    this.usedBuffers.get(key).push(buffer);
     this.numUsedBuffers++;
+    this.numBytesUsed += size;
 
-    if (this.freeBuffers.get(key).length > 0) {
-      this.numFreeBuffers--;
-
-      const newBuffer = this.freeBuffers.get(key).shift();
-      this.usedBuffers.get(key).push(newBuffer);
-      return newBuffer;
-    }
-
-    this.numBytesAllocated += size;
-    const newBuffer = this.device.createBuffer({size, usage, mappedAtCreation});
-    this.usedBuffers.get(key).push(newBuffer);
-
-    return newBuffer;
+    return buffer;
   }
 
-  releaseBuffer(buffer: GPUBuffer, size: number, usage: GPUBufferUsageFlags) {
+  releaseBuffer(
+      buffer: GPUBuffer, size: number, usage: GPUBufferUsageFlags,
```

Comment:
Can we remove `size: number, usage: GPUBufferUsageFlags`? Use `buffer.size`, `buffer.usage` when needed.

Reply:
This needs another PR to change bufferInfo to GPUBuffer.